public/app/features/provisioning/hooks/useCreateOrUpdateRepository.ts

```diff
@@ -9,7 +9,18 @@ export function useCreateOrUpdateRepository(name?: string) {
   const updateOrCreate = useCallback(
     (data: RepositorySpec) => {
       if (name) {
-        return update({ name, repository: { metadata: { name }, spec: data } });
+        return update({
+          name,
+          repository: {
+            metadata: {
+              name,
+              // TODO? -- replace with patch spec, so the rest of the metadata is not replaced?
+              // Can that support optimistic locking? (eg, make sure the RV is the same?)
+              finalizers: ['create', 'remove-orphan-resources'],
```

Comment:
previously the finalizers (and rest of the metadata) get removed with frontend updates.

We should either:
* patch spec (but keep optimistic locking?)
* hold onto the object we are updating, and send it in the post -- not just the spec